src/ipc_bridge.rs

```diff
@@ -15,46 +19,122 @@ pub struct ClientId {
     pub token: mio::Token,
 }
 
-struct Connection {
+// We abstract over unix and TCP connections. Since receiver and sender both get a copy of the
+// socket, we need to clone them. Since we store them in slab (which means the trait cannot be
+// sized), we have to return boxes too.
+trait MioStream: Send + io::Read + io::Write + mio::Evented {
+    fn try_clone(&self) -> io::Result<Box<MioStream>>;
+}
```

Comment:
This looks very similar to the `TryClone` trait above. The only difference is that this one doesn't use the `Sized` bound. Any reason for having both these? Won't they just clobber each other?